cli/src/pcluster/models/cluster.py

```diff
@@ -254,21 +254,33 @@ def config_presigned_url(self) -> str:
     @property
     def compute_fleet_status(self) -> ComputeFleetStatus:
         """Status of the cluster compute fleet."""
-        compute_fleet_status_manager = ComputeFleetStatusManager(self.name)
-        status = compute_fleet_status_manager.get_status()
-        if status == ComputeFleetStatus.UNKNOWN:
-            stack_status_to_fleet_status = {
-                "CREATE_IN_PROGRESS": ComputeFleetStatus.STARTING,
-                "DELETE_IN_PROGRESS": ComputeFleetStatus.STOPPING,
-                "CREATE_FAILED": ComputeFleetStatus.STOPPED,
-                "ROLLBACK_IN_PROGRESS": ComputeFleetStatus.STOPPING,
-                "ROLLBACK_FAILED": ComputeFleetStatus.STOPPED,
-                "ROLLBACK_COMPLETE": ComputeFleetStatus.STOPPED,
-                "DELETE_FAILED": ComputeFleetStatus.STOPPING,
-            }
-            return stack_status_to_fleet_status.get(self.status, status)
+        status, _ = self.compute_fleet_status_with_last_updated_time
         return status
 
+    @property
+    def compute_fleet_status_with_last_updated_time(self):
+        """Status of the cluster compute fleet and the last compute fleet status updated time."""
+        if self.stack.scheduler == "slurm":
+            compute_fleet_status_manager = ComputeFleetStatusManager(self.name)
+            status, last_updated_time = compute_fleet_status_manager.get_status_with_last_updated_time()
+            if status == ComputeFleetStatus.UNKNOWN:
+                stack_status_to_fleet_status = {
+                    "CREATE_IN_PROGRESS": ComputeFleetStatus.STARTING,
+                    "DELETE_IN_PROGRESS": ComputeFleetStatus.STOPPING,
+                    "CREATE_FAILED": ComputeFleetStatus.STOPPED,
+                    "ROLLBACK_IN_PROGRESS": ComputeFleetStatus.STOPPING,
+                    "ROLLBACK_FAILED": ComputeFleetStatus.STOPPED,
+                    "ROLLBACK_COMPLETE": ComputeFleetStatus.STOPPED,
+                    "DELETE_FAILED": ComputeFleetStatus.STOPPING,
+                }
+                status = stack_status_to_fleet_status.get(self.status, status)
+        else:  # scheduler is AWS Batch:
+            status = ComputeFleetStatus(
+                AWSApi.instance().batch.get_compute_environment_state(self.stack.batch_compute_environment)
```

Comment:
what happens if this API call fails. I can think of a few failure scenarios: 
1. compute_environment does not exist yet or anymore (stack creation/deletion) - shall we handle this similarly to how we handle it for Slurm?
2. Insufficient IAM permissions (covered by convert_error?)
3. Throttling for the API call (covered by convert_error?)
4. Problems with the AWS Batch service (covered by convert_error?)

Reply:
1. I added similar check on `IN_PROGRESS` stack status.
2~4. They are handled by convert_error

Reply:
not sure I can find where you are handling number 1

Reply:
Done by only allowing get compute fleet status from stack in CREATE COMPLETE, UPDATE COMPLETE, UPDATE ROLLBACK COMPLETE

Reply:
I'm still not sure we addressed number 1. When executing describe_compute_fleet_status with a batch cluster for a stack that is not in complete status the `AWSApi.instance().batch.get_compute_environment_state` is going to raise. What happens then? Shall we apply same logic as Slurm and return UNKNOWN?